tests/clients/redis/test_redis.py

```diff
@@ -10,29 +10,28 @@
 
 
 class RedisTests(BaseTests):
+    def setUp(self):
+        self.redis_instance_name = self.randstr()
+        self.redis_config_target_path = "/tmp/redis.conf"
+        self.port = randint(1000, 2000)
```

Comment:
IMHO, this range should'not include ports < 1024, also, it should not be random, might cause problems on different systems, let's choose a port which is not well known and used by any other possible server.

Reply:
I would suggest using a more reliable way to get a free port.
currently, I'm using this code in my unit tests for nettools
# getting a random free port on localhost
```
with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
        s.bind(("127.0.0.1", 0))
        port = s.getsockname()[1]
```
if you want a free port to bind to the local machine ipv6 address ()
```
with socket.socket(socket.AF_INET6, socket.SOCK_STREAM) as s:
        s.bind(("::1", 0))
        port = s.getsockname()[1]
```
we need to make it a separate function, maybe in nettools ?

Reply:
> > Outside tmux, python3.6
> > ```
> > ============================================================== short test summary info ==============================================================
> > FAILED tests/sals/process/test_process.py::ProcessTests::test_07_check_running - AssertionError: True is not false
> > FAILED tests/sals/process/test_process.py::ProcessTests::test_09_get_process_environ - AssertionError: 2 != 1 : tail didn't start
> > FAILED tests/sals/process/test_process.py::ProcessTests::test_10_get_filtered_pids - AssertionError: False is not true
> > FAILED tests/sals/process/test_process.py::ProcessTests::test_13_get_kill_process_by_port - AssertionError: False is not true
> > FAILED tests/sals/process/test_process.py::ProcessTests::test_15_get_kill_process_by_pids - AssertionError: 2 != 1
> > FAILED tests/sals/process/test_process.py::ProcessTests::test_20_get_ports_mapping - AssertionError: False is not true
> > ================================== 6 failed, 187 passed, 3 skipped, 14 deselected, 9 warnings in 111.02s (0:01:51) ==================================
> > Makefile:10: recipe for target 'unittests' failed
> > make: *** [unittests] Error 1
> > ```
> 
> the tmux one succeeded :D, now we may have the process isn't up yet, @AhmedHanafy725 gonna check about the speed/time limitations

sleep for a sec before checking the process? increase the wait time would be helpful too.

Reply:
@sameh-farouk  You can add this as a helper function in nettools to be used later.